File: sources/Xatz (Intent base)/me/xatzdevelopments/util/RandomUtils.java
package me.xatzdevelopments.util;

import java.util.*;

public class RandomUtils
{
    public static Random RANDOM;
    
    static {
        RandomUtils.RANDOM = new Random();
    }
    
    public static double RandomBetween(final double MinValue, final double MaxValue) {
        return Math.random() * (MaxValue - MinValue + 1.0) + MinValue;
    }
    
    public static byte[] nextBytes(final int count) {
        final byte[] result = new byte[count];
        RandomUtils.RANDOM.nextBytes(result);
        return result;
    }
    
    public static int nextInt(final int startInclusive, final int endExclusive) {
        return (endExclusive - startInclusive <= 0) ? startInclusive : (startInclusive + RandomUtils.RANDOM.nextInt(endExclusive - startInclusive));
    }
    
    public static long nextLong(final long startInclusive, final long endExclusive) {
        return (startInclusive == endExclusive) ? startInclusive : ((long)nextDouble((double)startInclusive, (double)endExclusive));
    }
    
    public static double nextDouble(final double startInclusive, final double endInclusive) {
        return (startInclusive == endInclusive) ? startInclusive : (startInclusive + (endInclusive - startInclusive) * RandomUtils.RANDOM.nextDouble());
    }
    
    public static float nextFloat(final float startInclusive, final float endInclusive) {
        return (startInclusive == endInclusive) ? startInclusive : (startInclusive + (endInclusive - startInclusive) * RandomUtils.RANDOM.nextFloat());
    }
}

File: sources/Kevin Reborn/kevin/utils/RandomUtils.kt
/*
 * This program is free software: you can redistribute it and/or modify
 *   it under the terms of the GNU General Public License as published by
 *   the Free Software Foundation, either version 3 of the License, or
 *   (at your option) any later version.
 *
 *   This program is distributed in the hope that it will be useful,
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *   GNU General Public License for more details.
 *
 *   You should have received a copy of the GNU General Public License
 *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package kevin.utils

import java.util.*

object RandomUtils {
    val random = Random()

    @JvmStatic
    fun nextInt(startInclusive: Int, endExclusive: Int): Int {
        return if (endExclusive - startInclusive <= 0) startInclusive else startInclusive + random.nextInt(endExclusive - startInclusive)
    }
    @JvmStatic
    fun nextDouble(startInclusive: Double, endInclusive: Double): Double {
        return if (startInclusive == endInclusive || endInclusive - startInclusive <= 0.0) startInclusive else startInclusive + (endInclusive - startInclusive) * Math.random()
    }
    @JvmStatic
    fun nextFloat(startInclusive: Float, endInclusive: Float): Float {
        return if (startInclusive == endInclusive || endInclusive - startInclusive <= 0f) startInclusive else (startInclusive + (endInclusive - startInclusive) * Math.random()).toFloat()
    }
    @JvmStatic
    fun randomNumber(length: Int): String {
        return random(length, "123456789")
    }

    @JvmStatic
    fun randomString(length: Int): String {
        return random(length, "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz")
    }
    @JvmStatic
    fun random(length: Int, chars: String): String {
        return random(length, chars.toCharArray())
    }
    @JvmStatic
    fun random(length: Int, chars: CharArray): String {
        val stringBuilder = StringBuilder()
        for (i in 0 until length) stringBuilder.append(chars[random.nextInt(chars.size)])
        return stringBuilder.toString()
    }
}